cmd/awscollector/Dockerfile

```diff
@@ -28,18 +28,12 @@ COPY .git /workspace/.git
 COPY Makefile /workspace/Makefile
 COPY config /workspace/config
 
+ARG TARGETPLATFORM
+RUN echo "Targate platform is $TARGETPLATFORM"
+
 #Copy binary if exist else build binary
-COPY go.* build/linux/aoc_linux_x86_6* /workspace/
-RUN if [[ -f /workspace/aoc_linux_x86_64 ]] ; \
-    then \
-        echo Copying binary \
-        && mkdir -p /workspace/build/linux \
-        && cp /workspace/aoc_linux_x86_64 /workspace/build/linux/aoc_linux_x86_64 ; \
-    else \
-        echo Building binary \
-        && go mod download  \
-        && make amd64-build ; \
-    fi
+COPY go.* build/linux/* /workspace/
+RUN cmd/awscollector/build_docker_binary.sh
```

Comment:
It isn't necessary to build the executable  again, is it?  The `docker-build` and `docker-build-arm` make targets already require the `amd64-build` and `arm64-build` targets, respectively.  We should simply copy the already-built binary into the final stage image.

This should also ensure that any packages we build from the binaries built from `a(md|rm)64-build`, such as RPM or DEP packages, are using the same binary as the container image.

Reply:
The idea was to keep backward compatibility where if we do not have the binary we build it. People can use docker commands not through the makefile if they want. Why are they doing this, not sure.

Reply:
What would be your recommendation? Just remove the ability to build the binary inside of the dockerfile if no binary is found?

Reply:
After speaking with @Aneurysm9 we should not build the binary inside the docker container since make will build the binary and in ci we cache the binary.